stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -4559,12 +4678,14 @@ impl SortitionDB {
         Ok(ret)
     }
 
+    /// DO NOT CALL FROM CONSENSUS CODE
```

Comment:
Why shouldn’t this function be invoked by consensus code? What do you mean by consensus code here?

Reply:
Specifically, because it calls `SortitionDB::get_canonical_sortition_tip()`, which can change whenever a new burnchain block is processed.  If this gets called from block-processing code in any places besides the burnchain block processor in the coordinator, then it's possible that different nodes will get different sortition IDs corresponding to different burnchain blocks while processing the same Stacks block.  That, in turn, can lead to a chain split.

Reply:
I see — can you make this comment more specific here, then?

```suggestion
    /// DO NOT call during Stacks block processing (including during Clarity VM evaluation). This function returns the latest data known to the node, which may not have been at the time of original block assembly.
```

Reply:
(I think the same comment applies to the rest of the DO NOT CALL comments)

Reply:
Sure